File: csharp/066_Plus_One.cs
public class Solution {
    public int[] PlusOne(int[] digits) {
        for (int i = digits.Length - 1; i >= 0; i--) {
            digits[i] += 1;
            if (digits[i] != 10) break;

            digits[i] = 0;
            if (i == 0) {
                int[] newArr = new int[digits.Length + 1];
                newArr[0] = 1;
                digits.CopyTo(newArr, 1);
                return newArr;
            }
        }

        return digits;
    }
}

File: cpp/066_Plus_One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        for (int i = digits.size() - 1; i >= 0; i--) {
            digits[i] += 1;
            if (digits[i] == 10) {
                if (i == 0) {
                    digits[i] = 0;
                    digits.insert(digits.begin(), 1);
                } else {
                    digits[i] = 0;
                }
            } else {
                break;
            }
        }

        return digits;
    }
};